eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/EidasSamlResource.java

```diff
@@ -75,4 +96,34 @@ private String getLevelsOfAssurance(AuthnRequest authnRequest) {
                 .map(AuthnContextClassRef::getAuthnContextClassRef)
                 .collect(Collectors.joining(","));
     }
+
+    private Credential getSigningCredential(String base64X509Certificate) throws CertificateException {
+        CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+
+        byte[] decodedCert = Base64.getMimeDecoder().decode(base64X509Certificate);
```

Comment:
Does this need to be a MIME decoder? I had to look it up. Looks like B64 with added formatting: https://www.tutorialspoint.com/java8/java8_base64.htm

Reply:
If you change it to a standard decoder lots of tests fail.  The MimeDecoder is the more robust approach and I'm sure it's used elsewhere in the Verify codebase (as that's where I first saw it) - we could alter whatever test data it finds offensive but by using it, we build in some robustness and keep [Jon Postel](https://en.wikipedia.org/wiki/Robustness_principle) happy.

Reply:
How is it more robust?

Reply:
If you change it to Base64.getDecoder()... you'll see this exception thrown in lots of tests:

```! java.lang.IllegalArgumentException: Illegal base64 character a```

The only thing that has changed is the decoder.  It might be that we'll never come across something with the offending encoding in practice but if we do, it will work.  For the record, I'm not sure why a would be an illegal character but the MimeDecoder handles input that the vanilla one doesn't.